tpm2_pytss/internal/crypto.py

```diff
@@ -345,34 +348,100 @@ def verify_signature_hmac(signature, key, data):
     sh.update(data)
     hdata = sh.finalize()
     sig = bytes(signature.signature.hmac)
-    h = hmac.HMAC(key, dt(), backend=default_backend())
+    h = HMAC(key, dt(), backend=default_backend())
     h.update(hdata)
     h.verify(sig)
 
 
-def verify_signature(signature, key, data):
+def _verify_signature(signature, key, data):
     if hasattr(key, "publicArea"):
         key = key.publicArea
     kt = getattr(key, "type", None)
-    if kt in (lib.TPM2_ALG_RSA, lib.TPM2_ALG_ECC):
+    if kt in (TPM2_ALG.RSA, TPM2_ALG.ECC):
         key = public_to_key(key)
-    if signature.sigAlg in (lib.TPM2_ALG_RSASSA, lib.TPM2_ALG_RSAPSS):
+    if signature.sigAlg in (TPM2_ALG.RSASSA, TPM2_ALG.RSAPSS):
         if not isinstance(key, rsa.RSAPublicKey):
             raise ValueError(
                 f"bad key type for {signature.sigAlg}, expected RSA public key, got {key.__class__.__name__}"
             )
         verify_signature_rsa(signature, key, data)
-    elif signature.sigAlg == lib.TPM2_ALG_ECDSA:
+    elif signature.sigAlg == TPM2_ALG.ECDSA:
         if not isinstance(key, ec.EllipticCurvePublicKey):
             raise ValueError(
                 f"bad key type for {signature.sigAlg}, expected ECC public key, got {key.__class__.__name__}"
             )
         verify_signature_ecc(signature, key, data)
-    elif signature.sigAlg == lib.TPM2_ALG_HMAC:
+    elif signature.sigAlg == TPM2_ALG.HMAC:
         if not isinstance(key, bytes):
             raise ValueError(
                 f"bad key type for {signature.sigAlg}, expected bytes, got {key.__class__.__name__}"
             )
         verify_signature_hmac(signature, key, data)
     else:
         raise ValueError(f"unsupported signature algorithm: {signature.sigAlg}")
+
+
+def generate_rsa_seed(
+    key: rsa.RSAPublicKey, hashAlg: int, label: bytes
+) -> Tuple[bytes, bytes]:
+    halg = _get_digest(hashAlg)
+    if halg is None:
+        raise ValueError(f"unsupported digest algorithm {hashAlg}")
+    seed = secrets.token_bytes(halg.digest_size)
+    mgf = padding.MGF1(halg())
+    padd = padding.OAEP(mgf, halg(), label)
+    enc_seed = key.encrypt(seed, padd)
+    return (seed, enc_seed)
+
+
+def generate_ecc_seed(
```

Comment:
Why are the 2 seed functions non-underscore versus underscore?

Is your convention here that things under internal are all for internal use, but functions/methods that start with an `_` are not general internal API. Ie something like underscore prefix functions are static functions for use in file scope and non-underscore prefix internal API is for general internal API consumption?

Reply:
I skipped them as they aren't imported anywhere, they are only called by _generate_seed.
I'll fix it when I rebase